mangarack-component-core/src/providers/dynastyreader/series.ts

```diff
@@ -0,0 +1,172 @@
+    type: mio.toSeriesType(getType(document), mio.SeriesType.Manga)
+  };
+}
+
+/**
+ * Promises the document.
+ * @param address The address.
+ * @return The promise for the document.
+ */
+async function downloadDocumentAsync(address: string): Promise<mio.IHtmlDocument> {
+  const body = await httpService().text(address, {}, mio.RequestType.TimeoutWithRetry).getAsync();
+  return htmlService().load(body);
+}
+
+/**
+ * Promises the image.
+ * @param $ The selector.
+ * @return The promise for the image.
+ */
+function downloadImageAsync($: mio.IHtmlDocument): Promise<mio.IBlob> {
+  const address = $('.cover img').attr('src');
+
+  if (address) {
+    return httpService().blob(`${providerDomain}${address}`, {}, mio.RequestType.TimeoutWithRetry).getAsync();
+  }
+
+  throw new Error('Invalid series cover address.');
+}
+
+/**
+ * Retrieves each artist.
+ * @return Each artist.
+ */
+function getArtists(): string[] {
+  return [];
+}
+
+/**
+ * Retrieves each author.
+ * @param $ The selector.
+ * @return Each author.
+ */
+function getAuthors($: mio.IHtmlDocument): string[] {
+  return $('.tag-title a[href*=\'/authors/\']')
+    .map((index, a) => $(a).text())
+    .get();
+}
+
+/**
+ * Retrieves Metadata from URL
+ * @param title The title.
+ * @return Each child.
+ */
+function getMetadata(title: string): mio.IChapterMetadata {
+  const numberMatch = title.match(/[-.0-9]+/);
+
+  return {
+    number: mio.option(numberMatch ? parseFloat(numberMatch[0]) : null),
+    title: title,
+    version: mio.option<number>(),
+    volume: mio.option<number>()
+  };
+}
+
+/**
+ * Retrieves each child.
+ * @param $ The selector.
+ * @return Each child.
+ */
+function getChapters($: mio.IHtmlDocument): mio.IChapter[] {
+  const results: mio.IChapter[] = [];
+
+  $('.chapter-list a[href*=\'/chapters/\']').map((index, a) => {
+    const address = $(a).attr('href');
+    const isValid = /[0-9.]+$/i.test(address);
+
+    if (address && isValid) {
+      const metadata = getMetadata($(a).text());
+      const url = `${providerDomain}${address}`;
+
+      results.push(createChapter(url, metadata));
+    }
+  });
+
+  return results;
+}
+
+/**
+ * Retrieves each genre.
+ * @param $ The selector.
+ * @return Each genre.
+ */
+function getGenres($: mio.IHtmlDocument): string[] {
+  return $('a.label')
+    .map((index, a) => $(a).text())
+    .get()
+    .map(value => remapGenreType[value] || value);
+}
+
+/**
+ * Retrieves the summary.
+ * @param $ The selector.
+ * @return The summary.
+ */
+function getSummary($: mio.IHtmlDocument): string {
+  return $('.description').text();
+}
+
+/**
+ * Retrieves the title.
+ * @param $ The selector.
+ * @return The title.
+ */
+function getTitle($: mio.IHtmlDocument): string {
+  return $('.tag-title b').text();
+}
+
+/**
+ * Retrieves the type.
+ * @param $ The selector.
+ * @return The type.
+ */
+function getType($: mio.IHtmlDocument): string {
+  const genres = getGenres($);
+  if (genres.indexOf('Art book') !== -1) {
```

Comment:
Like with the throws, making these explicit blocks make it very clear at one glance you've got 3 exits of this function. So, if, else if, else. Again, readability weirdness of mine.